doc/api/http2.md

```diff
@@ -39,6 +39,12 @@ server.on('stream', (stream, headers) => {
 server.listen(80);
 ```
 
+Note that this is an HTTP server and does not support HTTPS.
```

Comment:
s/this/the above example/

Also, I think using HTTP/HTTPS could be confusing since I would bet most people would associate those terms with HTTP 1.x.